site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -51,11 +51,102 @@ public function editGradeableRequest($gradeable_id, $nav_tab = 0) {
+            if ($config->isDiscussionBased()) {
+                $return_json['discussion_based'] = true;
+                $return_json['discussion_thread_id'] = $config->getDiscussionThreadId();
+            }
+            if ($config->isVcs()) {
+                $vcs_values = [];
+                switch ($config->getVcsHostType()) {
+                    case 0:
+                        $vcs_values['repository_type'] = 'submitty-hosted';
+                        break;
+                    case 1:
+                        $vcs_values['repository_type'] = 'submitty-hosted-url';
+                        $vcs_values['vcs_path'] = $config->getVcsPartialPath();
+                        break;
+                    case 2:
+                        $vcs_values['repository_type'] = 'public-github';
+                        break;
+                    case 3:
+                        $vcs_values['repository_type'] = 'private-github';
+                        break;
+                    case 4:
+                        $vcs_values['repository_type'] = 'self-hosted';
+                        $vcs_values['vcs_path'] = $config->getVcsPartialPath();
+                        break;
+                    default:
+                        JsonResponse::getFailResponse('Invalid VCS Type');
+                        break;
+                }
+                if ($config->isUsingSubdirectory()) {
+                    $vcs_values['subdirectory'] = $config->getVcsSubdirectory();
+                }
+                $return_json['vcs'] = $vcs_values;
+            }
+
+            $dates = [];
+            $dates['ta_view_start_date'] = $config->getTaViewStartDate()->format('Y-m-d H:i:s');
+            $dates['grade_start_date'] = $config->getGradeStartDate()->format('Y-m-d H:i:s');
+            $dates['grade_due_date'] = $config->getGradeDueDate()->format('Y-m-d H:i:s');
+            $dates['grade_released_date'] = $config->getGradeReleasedDate()->format('Y-m-d H:i:s');
+            $dates['team_lock_date'] = $config->getTeamLockDate()->format('Y-m-d H:i:s');
+            $dates['submission_open_date'] = $config->getSubmissionOpenDate()->format('Y-m-d H:i:s');
+            $dates['submission_due_date'] = $config->getSubmissionDueDate()->format('Y-m-d H:i:s');
+            $dates['grade_inquiry_start_date'] = $config->getGradeInquiryStartDate()->format('Y-m-d H:i:s');
+            $dates['grade_inquiry_due_date'] = $config->getGradeInquiryDueDate()->format('Y-m-d H:i:s');
+
+            $dates['has_due_date'] = $config->hasDueDate();
+            $dates['has_release_date'] = $config->hasReleaseDate();
+            $dates['late_submission_allowed'] = $config->isLateSubmissionAllowed();
+            $dates['late_days'] = $config->getLateDays();
+            $return_json['dates'] = $dates;
+        }
+
+        $this->core->getOutput()->useHeader(false);
+        $this->core->getOutput()->useFooter(false);
+        echo json_encode($return_json, JSON_PRETTY_PRINT);
```

Comment:
The `render()` function in `JsonResponse.php` essentially does the same thing, and would probably be a better option here.  The link I sent you to the example in `PlagiarismController.php` was probably not the best example in hindsight.